File: annotation/src/main/java/custom/eventbus/TwoActivity.java
package custom.eventbus;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.util.Log;
import android.view.View;

import com.example.annotation.R;

import java.lang.ref.Reference;
import java.lang.ref.WeakReference;

import custom.eventbus.annotation.Subscribe;

public class TwoActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        MethodManager methodManager = new MethodManager();
        WeakReference<MethodManager> str = new WeakReference<MethodManager>(methodManager);
        setContentView(R.layout.activity_two);
        EventBus.getInstance().register(this);
        findViewById(R.id.btn).setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                EventBus.getInstance().post("王菲");

                finish();
            }
        });
    }

    @Subscribe(value = ThreadMode.MAIN,stick = true)
    public void eventMainThread(Object obj){
        Log.e("TwoActivity",obj + "");
    }

    @Override
    protected void onDestroy() {
        super.onDestroy();
        EventBus.getInstance().unregister(this);
    }
}

File: app/src/main/java/com/example/myapplication/event/TwoActivity.kt
package com.example.myapplication.event

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.util.Log
import com.example.myapplication.R
import com.example.myapplication.util.PriorityRunnable
import com.example.myapplication.util.ThreadPoolManager
import kotlinx.android.synthetic.main.activity_two.*
import kotlin.math.E

class TwoActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_two)

        val threadPoolManager = ThreadPoolManager()

        // 前三个是由线程池设置的核心数量执行的，后面的是根据我们的优先级进行排序的
        for (c in 1..10) {
            val priority = c
            threadPoolManager.executorService.execute(object : PriorityRunnable(priority) {
                override fun doSth() {
                    val name = Thread.currentThread().name
                    Log.e("OneActivity", "线程 $name 正在执行优先级为 $priority 的任务")

                    try {
                        Thread.sleep(200)
                    } catch (e: InterruptedException) {
                        e.printStackTrace()
                    }
                }
            })
        }

        btn.setOnClickListener{
            EventBus.getInstance().post("tuige")
            finish()
        }
    }
}
